packages/ember-glimmer/lib/component-managers/capabilities.ts

```diff
@@ -0,0 +1,30 @@
+import { assert } from 'ember-debug';
+import { assign } from 'ember-utils';
+
+// because we're not yet compatible with latest Glimmer
+// we lack typings in Ember.js.
+// steal capabilities interface from http://bit.ly/2gMdhv5 for now
+export interface ComponentCapabilities {
+  dynamicLayout: boolean;
+  dynamicTag: boolean;
+  prepareArgs: boolean;
+  createArgs: boolean;
+  attributeHook: boolean;
+  elementHook: boolean;
+}
+
+// default capabilities values stolen from http://bit.ly/2zfZQyz
+export const DEFAULT_CAPABILITIES: ComponentCapabilities = {
+  dynamicLayout: true,
+  dynamicTag: true,
+  prepareArgs: true,
+  createArgs: true,
+  attributeHook: false,
+  elementHook: false
+};
+
+export default function capabilities(specifier: string, features: Object): Object {
+  assert('you need to pass in a specifier', specifier !== undefined);
+
+  return assign(DEFAULT_CAPABILITIES, features);
```

Comment:
Hmm, this seems wrong.  The `DEFAULT_CAPABILITIES` object will be mutated for each invocation.  I think you probably want:

```js
return assign({}, DEFAULT_CAPABILITIES, features);
```